packages/xo-web/src/xo-app/vm/tab-network.js

```diff
@@ -302,6 +309,241 @@ class VifStatus extends BaseComponent {
   }
 }
 
+// -----------------------------------------------------------------------------
+
+const USABLE_PORT_PROTOCOL = ['TCP', 'ICMP', 'UDP']
+class NewAclRuleForm extends BaseComponent {
```

Comment:
Prefer creating [Reaclette components](https://github.com/vatesfr/xen-orchestra/blob/60a696916bada66e682f2b1527097dd514fc0cd1/packages/xo-server/sample.config.toml#L3-L5), you can ask for help to XO devs like @badrAZ, @pdonias or @Rajaa-BARHTAOUI.